### PR TITLE
Add several UTs to improve drainer's coverage

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
[TOOL-1044](https://internal.pingcap.net/jira/browse/TOOL-1044)

### What is changed and how it works?
Five UTs are added in this PR. Coverage for `/drainer` is improved to around 77.1%.
1. [commit 1](https://github.com/lichunzhu/tidb-binlog/commit/8763780a6404243b972a7fe6e418314722a2c166) adds a rename table test in `schema_test.go`. Command `DROP TABLE` will use the new table name `RT`.
2. [commit 2](https://github.com/lichunzhu/tidb-binlog/commit/76731a0efc7e9eb1be1778ab9a0504c98dd7540e) adds close caller for merger and adds a UT check whether merger can be closed successfully in `merger.go` & `merger_test.go`.
3. [commit 3](https://github.com/lichunzhu/tidb-binlog/commit/d1141ac1a453d2dbe7ee2fcab9d452929b289b14) adds UT to test whether collector can update status successfully in `collector_test.go`.
4. [commit 4](https://github.com/lichunzhu/tidb-binlog/commit/9d948aa63e71fa76e80191eef91f1ef07a903d0d) adds UT to test whether config can generate correct kafka configs with zookeeper addrs or without zookeeper addrs in `config_test.go`.
5. [commit 5](https://github.com/lichunzhu/tidb-binlog/commit/09d241a965975ca65cb005f3c83f2a594ebd157d) adds UT to test whether `pump.PullBinlog` can successfully get binlog item and omit disordered items with wrong commitTS in `pump_test.go`.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has interface methods change
   Use `newZKFromConnectionString = zk.NewFromConnectionString` which changes the function usage in `/drainer/config.go`

Side effects

 - Increased code complexity

Related changes

### CODE REVIEW

File: drainer/collector_test.go [hunk 1]
```diff
@@ -463,3 +517,34 @@ func (s *keepUpdatingSuite) TestShouldUpdateWhenNotified(c *C) {
...
+	col := Collector{
+		merger:     &Merger{latestTS: 5, sources: map[string]MergeSource{}},
+		notifyChan: make(chan *notifyResult),
+		interval:   10 * time.Second,
+		reg:        r,
+		pumps:      map[string]*Pump{},
+		tiStore:    dummyStorage{},
+	}
+	var TS int64 = 5
+	err = col.updateStatus(ctx)
```
Comment (suzaku) [inline drainer/collector_test.go]: You can create a separate suite for test cases of `Collector.updateStatus`. This suite is for test cases of `Collector.keepUpdatingStatus`.

File: drainer/collector_test.go [hunk 2]
```diff
@@ -463,3 +517,34 @@ func (s *keepUpdatingSuite) TestShouldUpdateWhenNotified(c *C) {
...
+	defer cancel()
+	etcdClient := etcd.NewClient(testEtcdCluster.RandClient(), DefaultRootPath)
+	r := node.NewEtcdRegistry(etcdClient, time.Duration(5)*time.Second)
+	ns := &node.Status{
+		NodeID:  "test",
+		Addr:    "test",
+		State:   node.Online,
+		IsAlive: true,
+	}
+	err := r.UpdateNode(context.Background(), "pumps", ns)
```
Comment (suzaku) [inline drainer/collector_test.go]: It should be OK to reuse `ctx` here?

File: drainer/collector_test.go [hunk 3]
```diff
@@ -48,6 +49,59 @@ type dummyStore struct {
...
+
+func (ds dummyStorage) Close() error {
+	return nil
+}
+
+func (ds dummyStorage) UUID() string {
+	return ""
+}
+
+func (ds dummyStorage) CurrentVersion() (kv.Version, error) {
```
Comment (suzaku) [inline drainer/collector_test.go]: It seems only `CurrentVersion` is required in this case. You can try leaving the other dummy implementations out and see if it still works.

File: drainer/collector_test.go [hunk 4]
```diff
@@ -463,3 +517,34 @@ func (s *keepUpdatingSuite) TestShouldUpdateWhenNotified(c *C) {
...
+
+	col := Collector{
+		merger:     &Merger{latestTS: 5, sources: map[string]MergeSource{}},
+		notifyChan: make(chan *notifyResult),
+		interval:   10 * time.Second,
+		reg:        r,
+		pumps:      map[string]*Pump{},
+		tiStore:    dummyStorage{},
+	}
+	var TS int64 = 5
```
Comment (suzaku) [inline drainer/collector_test.go]: Where did this `5` come from?

File: drainer/config_test.go [hunk 5]
```diff
@@ -18,12 +18,16 @@ import (
...
+	"time"
 
 	"github.com/BurntSushi/toml"
 	"github.com/coreos/etcd/integration"
 	. "github.com/pingcap/check"
 	"github.com/pingcap/parser/mysql"
+	dsync "github.com/pingcap/tidb-binlog/drainer/sync"
 	"github.com/pingcap/tidb-binlog/pkg/util"
+	"github.com/pingcap/tidb-binlog/pkg/zk"
+	gozk "github.com/samuel/go-zookeeper/zk"
```
Comment (suzaku) [inline drainer/config_test.go]: According to the [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments#imports), we should rename the project-specific `zk`, not the more general one.

File: drainer/merge.go [hunk 6]
```diff
@@ -242,6 +242,7 @@ func (m *Merger) run() {
 		m.resetSourceChanged()
 
 		if m.isClosed() {
+			log.Info("merger is closed successfully")
```
Comment (suzaku) [inline drainer/merge.go]:
```suggestion
			log.Info("Merger is closed successfully")
```

File: drainer/merge_test.go [hunk 7]
```diff
@@ -135,3 +135,25 @@ func (s *testMergerSuite) merge(c *C, strategy string) {
...
+	for i := 0; i < sourceNum; i++ {
+		source := MergeSource{
+			ID:     strconv.Itoa(i),
+			Source: make(chan MergeItem),
+		}
+		sources[i] = source
+	}
+	merger := NewMerger(0, normalStrategy, sources...)
+	merger.Close()
+	time.Sleep(time.Second)
```
Comment (suzaku) [inline drainer/merge_test.go]: Is there anything we can do to avoid waiting for 1 second here?

File: drainer/pump_test.go [hunk 8]
```diff
@@ -15,7 +15,13 @@ package drainer
 
 import (
 	"context"
+	"time"
+
 	. "github.com/pingcap/check"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tipb/go-binlog"
+	pb "github.com/pingcap/tipb/go-binlog"
```
Comment (suzaku) [inline drainer/pump_test.go]: Duplicate imports?

File: drainer/pump_test.go [hunk 9]
```diff
@@ -40,3 +46,77 @@ func (s *pumpSuite) TestGetCompressorName(c *C) {
...
+			c.Assert(binlogItemEntity.GetCommitTs(), Equals, commitTs)
+		case <-time.After(time.Second):
+			c.Fatal("Haven't receive pump binlog item in 1 sec")
+		}
+	}
+	c.Assert(p.latestTS, Equals, commitTsArray[len(commitTsArray)-1])
+
+	// should omit disorder binlog item, latestTs should be 29
+	go func() {
+		for _, commitTs := range wrongCommitTsArray {
```
Comment (suzaku) [inline drainer/pump_test.go]: Create a helper function for sending binlogs?

File: drainer/pump_test.go [hunk 10]
```diff
@@ -40,3 +46,77 @@ func (s *pumpSuite) TestGetCompressorName(c *C) {
 	c.Assert(ok, IsTrue)
 	c.Assert(cp, Equals, "gzip")
 }
+
+var binlogBytesChan chan []byte
```
Comment (suzaku) [inline drainer/pump_test.go]: You can make this a field of `mockPumpPullBinlogsClient`.

**Comment by lichunzhu** [comment]
All comments have been accepted and relative parts have been revealed. PTAL. @suzaku

File: drainer/collector_test.go [hunk 11]
```diff
@@ -46,6 +46,12 @@ type dummyStore struct {
 	kv.Storage
 }
 
+func (ds dummyStore) CurrentVersion() (kv.Version, error) {
+	return kv.NewVersion(2), nil
+}
+
+var DefaultRootPath = "/tidb-binlog/v1"
```
[the remark this answers is not in the document]
Comment (lichunzhu) [reply drainer/collector_test.go]: Changed to node.DefaultRootPath

File: drainer/collector_test.go [hunk 12]
```diff
@@ -463,3 +469,38 @@ func (s *keepUpdatingSuite) TestShouldUpdateWhenNotified(c *C) {
 	s.waitForSignal(c, signal, "Notifies don't finish in time")
 	c.Assert(callCount, Equals, 10)
 }
+
+type updateStatusSuite struct{}
+
+var _ = Suite(&updateStatusSuite{})
+
+func (s *updateStatusSuite) TestUpdateStatus(c *C) {
```
[the remark this answers is not in the document]
Comment (lichunzhu) [reply drainer/collector_test.go]: Revised

**Pull request merged**